src/hotspot/os/posix/signals_posix.cpp

```diff
@@ -581,6 +597,8 @@ int JVM_HANDLE_XXX_SIGNAL(int sig, siginfo_t* info,
 {
   assert(info != NULL && ucVoid != NULL, "sanity");
 
+  InSignalHandlerMark ishm;
```

Comment:
Is this going to behave as expected if we actually crash during signal processing and so re-enter this routine with a signal mark already active?